### ISSUE TITLE
Ubuntu Workers are not created on Vsphere (allow_insecure: true)

### ISSUE THREAD

**Issue by mfahlandt** [kubermatic/kubermatic#6859]
**What happened**:

We created a Seed on Vsphere and gave the possibility to use two templates. The vcenter Endpoint mus be reached via the use insecure flag true for now.

It works with the flatcar image, but with the ubuntu image we run into certificate issues (ubuntu 20.04) 
If you describe the machine controller you also see no reference for allowing insecure.

Here is the Seed Config

```

apiVersion: kubermatic.k8s.io/v1
kind: Seed
metadata:
  name: seed-bgh
  namespace: kubermatic
spec:
  country: "DE"
  location: "Location"
  datacenters:
########### Vsphere
    prod:
      country: "DE"
      location: "PROD"
      spec:
        vsphere:
          allow_insecure: true
          cluster: "Cluster"
          datacenter: "DC"
          datastore: "Datastore"
          endpoint: "https://vsphereEP"
          root_path: "/"
          templates:
            flatcar: "/templates/flatcar-03-2021"
            ubuntu: "/templates/ubuntu-03-2021"
    
```

The strange thing however is that it is actually connecting to the vcenter but complaining on the logout post.

```
I0408 16:15:49.815058       1 helper.go:187] Increasing disk size to 10 GB 

I0408 16:15:50.555769       1 helper.go:296] Uploading userdata ISO to datastore Datastore:datastore-588828 @ /DC/datastore/Datastore, destination iso is vigilant-franklin-5d59668b4c-5j7ns/cloud-init.iso 

I0408 16:15:51.320830       1 helper.go:300] Uploaded ISO file /tmp/vigilant-franklin-5d59668b4c-5j7ns.iso 

E0408 16:15:51.367419       1 client.go:75] vsphere client failed to logout: Post "https://Vsphere-center/sdk": x509: certificate signed by unknown authority 

I0408 16:15:51.566687       1 helper.go:187] Increasing disk size to 10 GB 

I0408 16:15:51.763247       1 helper.go:187] Increasing disk size to 10 GB 

I0408 16:15:52.144215       1 helper.go:296] Uploading userdata ISO to datastore Datastore:datastore-588828 @ /vDC/datastore/DCDatastore, destination iso is vigilant-franklin-5d59668b4c-z45ns/cloud-init.iso 

I0408 16:15:52.316501       1 helper.go:296] Uploading userdata ISO to datastore Datastore:datastore-588828 @ /vDC/datastore/DCDatastore, destination iso is vigilant-franklin-5d59668b4c-wg7mm/cloud-init.iso 

I0408 16:15:52.510901       1 provider.go:441] Successfully destroyed vm vigilant-franklin-5d59668b4c-5j7ns 

E0408 16:15:52.517180       1 machine_controller.go:363] Failed to reconcile machine "vigilant-franklin-5d59668b4c-5j7ns": failed to create machine at cloudprovider, due to failed to destroy vm vigilant-franklin-5d59668b4c-5j7ns after failing upload and attach userdata iso: failed to get devices: Post "https://Vsphere-center/sdk": x509: certificate signed by unknown authority / Post "https://Vsphere-center/sdk": x509: certificate signed by unknown authority 

I0408 16:15:52.743974       1 helper.go:300] Uploaded ISO file /tmp/vigilant-franklin-5d59668b4c-z45ns.iso 

E0408 16:15:52.796452       1 client.go:75] vsphere client failed to logout: Post "https://Vsphere-center/sdk": x509: certificate signed by unknown authority 

I0408 16:15:52.902677       1 helper.go:300] Uploaded ISO file /tmp/vigilant-franklin-5d59668b4c-wg7mm.iso 

E0408 16:15:52.950884       1 client.go:75] vsphere client failed to logout: Post "https://Vsphere-center/sdk": x509: certificate signed by unknown authority 

I0408 16:15:54.125677       1 provider.go:441] Successfully destroyed vm vigilant-franklin-5d59668b4c-z45ns 

E0408 16:15:54.135483       1 machine_controller.go:363] Failed to reconcile machine "vigilant-franklin-5d59668b4c-z45ns": failed to create machine at cloudprovider, due to failed to destroy vm vigilant-franklin-5d59668b4c-z45ns after failing upload and attach userdata iso: failed to get devices: Post "https://Vsphere-center/sdk": x509: certificate signed by unknown authority / Post "https://Vsphere-center/sdk": x509: certificate signed by unknown authority 

I0408 16:15:54.227034       1 provider.go:441] Successfully destroyed vm vigilant-franklin-5d59668b4c-wg7mm 

E0408 16:15:54.233928       1 machine_controller.go:363] Failed to reconcile machine "vigilant-franklin-5d59668b4c-wg7mm": failed to create machine at cloudprovider, due to failed to destroy vm vigilant-franklin-5d59668b4c-wg7mm after failing upload and attach userdata iso: failed to get devices: Post "https://Vsphere-center/sdk": x509: certificate signed by unknown authority / Post "https://Vsphere-center/sdk": x509: certificate signed by unknown authority 
```

**Comment by mate4st** [comment]
Potential bugfix:

https://github.com/kubermatic/machine-controller/pull/979
https://github.com/kubermatic/kubermatic/pull/7077

**Comment by toschneck** [comment]
@WeirdMachine tested the bugfix and they worked as expected. cc @shaase-ctrl please include the MC fix also into 2.17.1

**Comment by mate4st** [comment]
https://github.com/kubermatic/machine-controller/pull/996
https://github.com/kubermatic/kubermatic/pull/7201
